jooq/src/main/java/com/cerner/beadledom/jooq/ThreadLocalCommitHookExecutingTransactionListener.java

```diff
@@ -0,0 +1,117 @@
+package com.cerner.beadledom.jooq;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import org.jooq.TransactionContext;
+import org.jooq.TransactionListener;
+import org.jooq.impl.DefaultTransactionListener;
+
+/**
+ * An implementation of {@link TransactionListener} that sets the current transaction for
+ * {@link JooqTransactionalHooks} and removes hooks on rollback or executes commit hooks when the
+ * top-level transaction is committed.
+ *
+ * @author John Leacox
+ * @since 3.3
+ */
+class ThreadLocalCommitHookExecutingTransactionListener extends DefaultTransactionListener {
+  private final ThreadLocal<Deque<JooqTransaction>> transactionDeques = new ThreadLocal<>();
+
+  // This assumes that the hooks implementation is using ThreadLocal or some other means to have
+  // the same lifecycle as this listener.
+  private final JooqTransactionalHooks transactionalHooks;
```

Comment:
Why would this type be anything other than ThreadLocalJooqTransactionalHooks? Maybe my question shines a light on the fact that I don't completely understand this disclaimer

Reply:
I removed this comment with the change to an interface: https://github.com/cerner/beadledom/pull/140/commits/ed118b69f588ab7f1fe9de2e975b4236cdd2e7a2